modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/portlet/action/DeleteCustomDataSetMVCResourceCommand.java

```diff
@@ -0,0 +1,101 @@
+ */
+
+package com.liferay.frontend.data.set.admin.web.internal.portlet.action;
+
+import com.liferay.frontend.data.set.admin.web.internal.constants.FDSAdminPortletKeys;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.rest.dto.v1_0.ObjectEntry;
+import com.liferay.object.rest.manager.v1_0.DefaultObjectEntryManagerProvider;
+import com.liferay.object.rest.manager.v1_0.ObjectEntryManager;
+import com.liferay.object.rest.manager.v1_0.ObjectEntryManagerRegistry;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseTransactionalMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+import com.liferay.portal.vulcan.pagination.Page;
+
+import jakarta.portlet.ResourceRequest;
+import jakarta.portlet.ResourceResponse;
+
+import java.util.Collection;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Antonio Ortega
+ */
+@Component(
+	property = {
+		"jakarta.portlet.name=" + FDSAdminPortletKeys.FDS_ADMIN,
+		"mvc.command.name=/frontend_data_set_admin/delete_custom_data_set"
+	},
+	service = MVCResourceCommand.class
+)
+public class DeleteCustomDataSetMVCResourceCommand
+	extends BaseTransactionalMVCResourceCommand {
+
+	@Override
+	protected void doTransactionalCommand(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)resourceRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.
+				fetchObjectDefinitionByExternalReferenceCode(
+					"L_DATA_SET_SNAPSHOT", themeDisplay.getCompanyId());
+
+		long fdsId = ParamUtil.getLong(resourceRequest, "id");
+
+		String fdsName = ParamUtil.getString(
+			resourceRequest, "externalReferenceCode");
+
+		ObjectEntryManager objectEntryManager =
+			DefaultObjectEntryManagerProvider.provide(
+				_objectEntryManagerRegistry.getObjectEntryManager(
+					objectDefinition.getCompanyId(),
+					objectDefinition.getStorageType()));
+
+		Page<ObjectEntry> page = objectEntryManager.getObjectEntries(
+			PortalUtil.getCompanyId(resourceRequest), objectDefinition, null,
+			null,
+			new DefaultDTOConverterContext(
+				false, null, null, null, null,
+				LocaleUtil.getMostRelevantLocale(), null, null),
+			StringBundler.concat(
+				"(fdsName eq '", fdsName, "' and creatorId eq ",
```

Comment:
It's odd to me to see this kind of low level code in an `MVCResourceCommand`. I'd expect to use a getter that takes `fdsName` and `creatorId` as arguments. But if this is the only way, and it works, :+1: